### DISCUSSION TITLE
Version 2.0

### DISCUSSION THREAD

**Discussion by sebastianwessel** [sebastianwessel/quickjs#30] [Announcements]
# Version 2.0

This version marks a big milestone. The goal is, to provide a stable, production ready version.

The version 2.0 will have a slightly different style, to improve overall usage and DX - something like this:

```typescript
const runtime = await loadQuickJs()

const sandboxOptions = {...}

// create a sandbox 
const finalResult = await runtime.runSandboxed(async(sandbox)=>{
  const step1Result = sandbox.evalCode('export default 1+1')
  // ... some other operation on host side
  const step2Result = sandbox.evalCode('export default 2+2')
  // ...something else
  
  // retuning a final result
  return 'The Final Result'

}, sandboxOptions)

// sandbox is now destroyed and can no longer be used

console.log(finalResult)
```

**Comment by digipigeon** [comment]
It would be good if this context is possible to run evalCode on an event-driven paradigm from within the `runtime.runSandboxed`

It would also be nice if this open context could be destroyed externally if required.

**Reply by sebastianwessel** [reply]
What exactly do you mean with:
> to run evalCode on an event-driven paradigm from within the runtime.runSandboxed

Depending on the general application needs, there are already options to destroy from outside. But it should be the very last option.

**Comment by sebastianwessel** [comment]
Version 2 is out!

It brings a lot of improvements and cool features.
Also, there is now a real website with a lot of documentation, and use case scenarios.

Read the announcement:

[What is new - Blog post](https://sebastianwessel.github.io/quickjs/blog/2024-09-01-version-2.html)
